include/iamclient/permservicehandler.hpp

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef PERMSERVICEHANDLER_HPP_
+#define PERMSERVICEHANDLER_HPP_
+
+#include <chrono>
+#include <memory>
+#include <string>
+
+#include <aos/iam/permhandler.hpp>
+
+#include "publicservicehandler.hpp"
+
+namespace aos::common::iamclient {
+/**
+ * Permissions service handler.
+ */
+class PermissionsServiceHandler : public iam::permhandler::PermHandlerItf {
+public:
+    /**
+     * Initializes permissions service handler.
+     *
+     * @param IAMProtectedServerURL IAM protected server URL.
+     * @param certStorage certificate storage.
+     * @param certProvider certificate provider.
+     * @return Error.
+     */
+    Error Init(
+        const std::string& IAMProtectedServerURL, const std::string& certStorage, TLSCredentialsItf& certProvider);
+
+    /**
+     * Adds new service instance and its permissions into cache.
+     *
+     * @param instanceIdent instance identification.
+     * @param instancePermissions instance permissions.
+     * @returns RetWithError<StaticString<cSecretLen>>.
+     */
+    RetWithError<StaticString<iam::permhandler::cSecretLen>> RegisterInstance(const InstanceIdent& instanceIdent,
+        const Array<iam::permhandler::FunctionalServicePermissions>& instancePermissions) override;
+
+    /**
+     * Unregisters instance deletes service instance with permissions from cache.
+     *
+     * @param instanceIdent instance identification.
+     * @returns Error.
+     */
+    Error UnregisterInstance(const InstanceIdent& instanceIdent) override;
+
+    /**
+     * Returns instance ident and permissions by secret and functional server ID.
+     *
+     * @param secret secret.
+     * @param funcServerID functional server ID.
+     * @param[out] instanceIdent result instance ident.
+     * @param[out] servicePermissions result service permission.
+     * @returns Error.
+     */
+    Error GetPermissions(const String& secret, const String& funcServerID, InstanceIdent& instanceIdent,
+        Array<iam::permhandler::PermKeyValue>& servicePermissions) override;
+
+private:
+    static constexpr auto cIAMPermissionsServiceTimeout = std::chrono::seconds(10);
+
+    std::shared_ptr<grpc::ChannelCredentials> CreateCredentials();
+
+    TLSCredentialsItf* mCertProvider {};
```

Comment:
Consider renaming to something like `mTLSCredentials` for example.

Reply:
Done